File: HackerRank Solutions/Algorithms/Implementation/Angry Professor/Angry Professor.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the angryProfessor function below.
def angryProfessor(k, a):
    StudentsArrivalTime = 0
    for i in range(len(a)):
        if(a[i] <= 0):
            StudentsArrivalTime += 1
        if(StudentsArrivalTime >= k):
            return "NO"
    return "YES"
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        nk = input().split()

        n = int(nk[0])

        k = int(nk[1])

        a = list(map(int, input().rstrip().split()))

        result = angryProfessor(k, a)

        fptr.write(result + '\n')

    fptr.close()

File: HackerRank Solutions/Problem Solving/Algorithms/Implementation/Angry Professor/Angry Professor.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int t, StudentsArrivalTime;
    cin >> t;
    for (int a0=0; a0<t; a0++){
        int n, k;
        cin >> n >> k;
        vector<int>a(n);
        StudentsArrivalTime = 0;
        for (int i=0; i<n; i++){
            cin >> a[i];
            if (a[i] <= 0){
                StudentsArrivalTime++;
            }
        }
        if (StudentsArrivalTime >= k){
            cout << "NO" << endl;
        } else{
            cout << "YES" << endl;
        }
    }
    return 0;
}

